01_fib/patrickmarabeas/main.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// Fibonacci slice generator
+// - Starts from 0
+// - Handles both positive and negative numbers
+// - Returns nil for input of 0
+// - Returns an error if the input would generate numbers outside of an int lower/upper values
+func calculateFibSequence(n int) ([]int, error) {
+	// Multiplied against the initial Fibonacci value (1) to spawn either a positive or negative sequence
+	sign := 1
+	// Early bailout returning zero value of a slice
+	if n == 0 {
+		return nil, nil
+	}
+	// Early bailout and error for return values outside of int limits
+	if n > 92 || n < -92 {
+		return nil,
+			fmt.Errorf(fmt.Sprintf(`ERROR: %d is outside of limits Please enter a number between -92 and 92`, n*sign))
```

Comment:
The `fmt.Sprintf` is unnecessary. `fmt.Errorf` does that already (the `f` suffix denotes a formatted string)

Reply:
Resolved.